test/conformance/kernel/urKernelSetArgSampler.cpp

```diff
@@ -5,6 +5,65 @@
 
 #include <uur/fixtures.h>
 
+struct urKernelSetArgSamplerTestWithParam
+    : uur::urBaseKernelTestWithParam<uur::SamplerCreateParamTy> {
+    void SetUp() {
+        program_name = "image_copy";
+        UUR_RETURN_ON_FATAL_FAILURE(
+            uur::urBaseKernelTestWithParam<uur::SamplerCreateParamTy>::SetUp());
+        // Images and samplers are not available on AMD
+        ur_platform_backend_t backend;
+        ASSERT_SUCCESS(urPlatformGetInfo(platform, UR_PLATFORM_INFO_BACKEND,
+                                         sizeof(backend), &backend, nullptr));
+        if (backend == UR_PLATFORM_BACKEND_HIP) {
+            GTEST_SKIP() << "Sampler are not supported on hip.";
+        }
+        UUR_RETURN_ON_FATAL_FAILURE(
+            uur::urBaseKernelTestWithParam<uur::SamplerCreateParamTy>::Build());
+
+        const auto param = getParam();
+        const auto normalized = std::get<0>(param);
+        const auto addr_mode = std::get<1>(param);
+        const auto filter_mode = std::get<2>(param);
+
+        ur_sampler_desc_t sampler_desc = {
+            UR_STRUCTURE_TYPE_SAMPLER_DESC, /* sType */
+            nullptr,                        /* pNext */
+            normalized,                     /* normalizedCoords */
+            addr_mode,                      /* addressingMode */
+            filter_mode                     /* filterMode */
+        };
+        ASSERT_SUCCESS(urSamplerCreate(context, &sampler_desc, &sampler));
+    }
+
+    void TearDown() {
+        if (sampler) {
+            ASSERT_SUCCESS(urSamplerRelease(sampler));
+        }
+        UUR_RETURN_ON_FATAL_FAILURE(uur::urBaseKernelTestWithParam<
+                                    uur::SamplerCreateParamTy>::TearDown());
+    }
+
+    ur_sampler_handle_t sampler = nullptr;
+};
+
+UUR_TEST_SUITE_P(
+    urKernelSetArgSamplerTestWithParam,
+    ::testing::Combine(
+        ::testing::Values(true, false),
```

Comment:
I've seen in other test suites for parameterising tests with enabling/disabling of flags the use of `testing::ValuesIn(uur::BoolTestParam::makeBoolParam("UsePool")` which could maybe could be of use here and with the test name printing function.

Reply:
I think `makeBoolParam` was used mainly to generate meaningful names for tests that we use true and false for the param only like `usepool_true` / `usepool_false`. But in our case `normalized` and `unnormalized` name should be more clear I think, so I modified it to reflect that in the device printer.

Reply:
Fair enough!